clash-prelude/src/Clash/Sized/BitVector.hs

```diff
@@ -25,9 +25,40 @@ module Clash.Sized.BitVector
   , bLit
     -- ** Concatenation
   , (++#)
+    -- * Modification
+  , (+>>.)
+  , (.<<+)
     -- ** Pattern matching
   , bitPattern
   )
 where
 
 import Clash.Sized.Internal.BitVector
+import Clash.Sized.Index (Index)
+import Clash.Class.BitPack (replaceBit)
+import Data.Bits (shiftL, shiftR)
+import GHC.TypeNats (KnownNat)
+
+infixr 4 +>>.
+-- | Shift in a bit from the MSB side of a 'BitVector'. Equal to right shifting
+-- the 'BitVector' by one and replacing the MSB with the bit to be shifted in.
+--
+-- >>> 1 +>>. 0b1111_0000 :: BitVector 8
+-- 0b1111_1000
+-- >>> 0 +>>. 0b1111_0000 :: BitVector 8
+-- 0b0111_1000
+--
+(+>>.) :: forall n. KnownNat n => Bit -> BitVector n -> BitVector n
+b +>>. bv = replaceBit @_ @(Index n) maxBound b (shiftR bv 1)
```

Comment:
I dislike using `shiftR`, `shiftL` because it clutters HDL with 64 bit wide numbers. If possible for free I'd rather see that avoided. An implementation would also be `pack . (b +>>)  . unpack` although that might be slower in simulation.

Reply:
We could define `shiftL1` and `shiftR1` being equal to 
```
(`shiftL` 1) / (`shiftR` 1)
```
 with much less bounds checking and stuff.

Reply:
And yes, sim speed for the route through `Vec n Bit` is poor.

Reply:
Are you worried about synthesis results @rowanG077, or just code clutter?

Reply:
I think you should also use `replaceBit#` directly because the path through `replaceBit` introduces an unnecessary `unsafeDupablePerfomIO` to catch `XException`s.

Reply:
In any case, I think we should make the black boxes for `shiftR#` and `shiftL#` (+ other operations having the same problem) special case on constants. I'll post an issue.

Reply:
@martijnbastiaan Not especially. I would be surprised if the extra wires wouldn't be removed. But more that we should at least try to make HDL "good". Unless there is a pretty good reason not to. (And a pretty good reason could be a lot of work or simulation performance.)

Reply:
The proposed way would be wildly inefficient in simulation yeah. But mostly I feel like this is a separate issue anyway: https://github.com/clash-lang/clash-compiler/issues/2308.

Reply:
Since we have the issue to track it I'm fine with this PR as is.